website/src/layouts/base/header/HamburgerMenuItem.astro

```diff
@@ -0,0 +1,63 @@
+---
+type Props = {
+    content:
+        | {
+              type: 'item';
+              href?: string;
+              externalLink?: boolean;
+          }
+        | {
+              type: 'section';
+              navigationEntries: {
+                  label: string;
+                  href?: string;
+                  externalLink?: boolean;
+              }[];
+          };
+    style: 'toplevel' | 'subitem';
+};
+
+const { content, style } = Astro.props;
+
+const liClass =
+    style === 'toplevel'
+        ? `text-xl font-bold ml-4 ${content.type === 'item' ? 'py-4' : ''}`
+        : 'text-md font-light ml-8 pt-2';
+---
+
+<li class={liClass}>
+    {
+        content.type === 'item' ? (
+            content.href === undefined ? (
+                <slot />
+            ) : (
+                <a
+                    class={
+                        content.externalLink === true
+                            ? 'after:iconify after:ml-0.5 after:align-text-bottom after:mdi--external-link'
+                            : ''
+                    }
+                    href={content.href}
+                >
+                    <slot />
+                </a>
+            )
+        ) : (
```

Comment:
It would be easier, to follow, when this is its own astro file. Then we have "HamburgerMenuItem" and "HamburgerMenuSection". They don't share so much here.